File: src/SnapBrief.Core/Exporting/PromptGenerator.cs
using System.Text;
using SnapBrief.Core.Models;

namespace SnapBrief.Core.Exporting;

public sealed class PromptGenerator
{
    public string Generate(SnapBriefSession session)
    {
        SessionValidation.Validate(session);
        var sections = new List<string>();

        if (ExportText.HasContent(session.GlobalNote))
        {
            sections.Add($"Общее пожелание:\n{session.GlobalNote}");
        }

        for (var captureIndex = 0; captureIndex < session.Captures.Length; captureIndex++)
        {
            var capture = session.Captures[captureIndex];
            var captureLabel = CaptureLabels.ForIndex(captureIndex);
            var section = new StringBuilder($"Снимок {captureLabel}");
            if (ExportText.HasContent(capture.Title))
            {
                section.Append(" — ").Append(capture.Title);
            }

            section.Append('.');
            if (ExportText.HasContent(capture.Note))
            {
                section.Append("\nКомментарий к снимку:\n").Append(capture.Note);
            }

            var labeledAnnotations = CaptureLabels.ForNotedAnnotations(captureLabel, capture).ToArray();
            var labelsById = labeledAnnotations.ToDictionary(item => item.Annotation.Id, item => item.DisplayLabel);
            foreach (var labeled in labeledAnnotations)
            {
                section.Append('\n')
                    .Append(labeled.DisplayLabel)
                    .Append(": ")
                    .Append(labeled.Annotation.Note);
                if (labeled.Annotation.ParentAnnotationId is { } parentId)
                {
                    var parentLabel = labelsById.GetValueOrDefault(parentId);
                    if (!string.IsNullOrEmpty(parentLabel)) section.Append(" (к области ").Append(parentLabel).Append(')');
                }
            }

            sections.Add(section.ToString());
        }

        return string.Join("\n\n", sections);
    }
}

File: macos/Sources/SnapBriefCore/Exporting/PromptGenerator.swift
import Foundation

/// Port of `src/SnapBrief.Core/Exporting/PromptGenerator.cs`. UI strings are Russian verbatim
/// (matches the app's Russian-first prompt text; `UiLanguage` only translates chrome, not this
/// generated document).
public struct PromptGenerator {
    public init() {}

    public func generate(_ session: SnapBriefSession) throws -> String {
        try SessionValidation.validate(session)
        var sections: [String] = []

        if ExportText.hasContent(session.globalNote) {
            sections.append("Общее пожелание:\n\(session.globalNote)")
        }

        for (captureIndex, capture) in session.captures.enumerated() {
            let captureLabel = try CaptureLabels.forIndex(captureIndex)
            var section = "Снимок \(captureLabel)"
            if ExportText.hasContent(capture.title) {
                section += " — \(capture.title)"
            }
            section += "."

            if ExportText.hasContent(capture.note) {
                section += "\nКомментарий к снимку:\n\(capture.note)"
            }

            for labeled in CaptureLabels.forNotedAnnotations(captureLabel: captureLabel, capture: capture) {
                section += "\n\(labeled.displayLabel): \(labeled.annotation.note)"
            }

            sections.append(section)
        }

        return sections.joined(separator: "\n\n")
    }
}
